server/src/main/java/timely/api/model/Metric.java

```diff
@@ -109,7 +122,9 @@ public void setVisibility(ColumnVisibility visibility) {
 
     public Mutation toMutation() {
         final byte[] row = rowCoder.encode(new ComparablePair<String, Long>(this.metric, this.timestamp));
-        final Value value = new Value(valueCoder.encode(this.value));
+        WRITE_BUFFER.get().clear();
+        WRITE_BUFFER.get().putDouble(this.value);
+        final Value value = new Value(WRITE_BUFFER.get().array());
```

Comment:
Doesn't this just get the backing array for the byte buffer? If the buffer has been reused and the array is bigger than the size of the data in it, then you'll be putting too much in the Value object.